### PR TITLE
Remove dependencies from tests that require additional libraries

### PR DESCRIPTION
**Pull request by peterbae**
Remove dependencies from tests that are from outside required libraries.

Replaced:

org.apache.commons.lang3.RandomStringUtils
org.apache.commons.codec.binary.Hex
org.apache.commons.codec.DecoderException;

And replaced with with other methods / java.util functions.

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/testframework/sqlType/SqlDateTime2.java [hunk 1]
```diff
@@ -48,7 +47,19 @@ public SqlDateTime2() {
...
+        
+        Random rnd = new Random();
+        // this will generate a number that is this.precision number of digits.
+        int randomNumeric;
+        if (this.precision <= 0) {
+            randomNumeric = 0;
+        } else if (this.precision == 1) {
+            randomNumeric = rnd.nextInt(10);
+        } else {
+            randomNumeric = (int) Math.pow(10, this.precision - 1) + rnd.nextInt((int) (9 * Math.pow(10, this.precision - 1)));
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/testframework/sqlType/SqlDateTime2.java]: can we save this to a var so we don't have to call Math.pow(10, this.precision - 1) twice?

File: src/test/java/com/microsoft/sqlserver/testframework/sqlType/SqlTime.java [hunk 2]
```diff
@@ -50,7 +49,18 @@ public SqlTime() {
...
+        // this will generate a number that is this.scale number of digits.
+        int randomNumeric;
+        if (this.scale <= 0) {
+            randomNumeric = 0;
+        } else if (this.scale == 1) {
+            randomNumeric = rnd.nextInt(10);
+        } else {
+            randomNumeric = (int) Math.pow(10, this.scale - 1) + rnd.nextInt((int) (9 * Math.pow(10, this.scale - 1)));
+        }
+
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/testframework/sqlType/SqlTime.java]: this is the same as above can we move this out to a method which generates n random digits?

File: src/test/java/com/microsoft/sqlserver/testframework/DBTable.java [hunk 3]
```diff
@@ -474,4 +472,11 @@ boolean passDataAsHex(int colNum) {
         return (JDBCType.BINARY == getColumn(colNum).getJdbctype() || JDBCType.VARBINARY == getColumn(colNum).getJdbctype()
                 || JDBCType.LONGVARBINARY == getColumn(colNum).getJdbctype());
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/testframework/DBTable.java]: I know this didn't get changed, but would be better if we just do getColumn(colNum).getJdbctype() once

File: src/test/java/com/microsoft/sqlserver/testframework/sqlType/SqlDateTime.java [hunk 4]
```diff
@@ -41,4 +42,22 @@ public SqlDateTime() {
     public Object createdata() {
         return new Timestamp(ThreadLocalRandom.current().nextLong(((Timestamp) minvalue).getTime(), ((Timestamp) maxvalue).getTime()));
     }
+    
+    protected int generateRandomInt(int length) {
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/testframework/sqlType/SqlDateTime.java]: comment header?

File: src/test/java/com/microsoft/sqlserver/testframework/sqlType/SqlDateTime.java [hunk 5]
```diff
@@ -41,4 +42,22 @@ public SqlDateTime() {
...
+        Random rnd = new Random();
+        int power = (int) Math.pow(10, length - 1);
+        
+        //Example of how this works:
+        // if length is 3, then we add 100 + random number between 0~899, so that we get a random number between 100~999.
+        int randomNumeric;
+        if (length <= 0) {
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/testframework/sqlType/SqlDateTime.java]: nitpicking....
we normally do (0 >= blah) etc...

File: src/test/java/com/microsoft/sqlserver/testframework/sqlType/SqlDateTime.java (hunk 4 above)
Comment (peterbae) [reply src/test/java/com/microsoft/sqlserver/testframework/sqlType/SqlDateTime.java]: nah we don't need it

File: src/test/java/com/microsoft/sqlserver/testframework/DBTable.java (hunk 3 above)
Comment (peterbae) [reply src/test/java/com/microsoft/sqlserver/testframework/DBTable.java]: cool, done

File: src/test/java/com/microsoft/sqlserver/testframework/sqlType/SqlDateTime.java (hunk 5 above)
Comment (peterbae) [reply src/test/java/com/microsoft/sqlserver/testframework/sqlType/SqlDateTime.java]: i think that's only for comparing to null (for good reasons)

**Pull request merged**